include/nbl/builtin/hlsl/device_capabilities_traits.hlsl

```diff
@@ -7,15 +7,1323 @@
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetDynamicOffsetSSBOs, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetImages, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetStorageImages, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxDescriptorSetInputAttachments, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxVertexOutputComponents, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationGenerationLevel, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationPatchSize, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlPerVertexInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlPerVertexOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlPerPatchOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationControlTotalOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationEvaluationInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTessellationEvaluationOutputComponents, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryShaderInvocations, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryInputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryOutputComponents, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryOutputVertices, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxGeometryTotalOutputComponents, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentInputComponents, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentOutputAttachments, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentDualSrcAttachments, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxFragmentCombinedOutputResources, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeSharedMemorySize, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCount[0], uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCountY[1], uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupCountZ[2], uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxComputeWorkGroupInvocations, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[0], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[1], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[2], uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(subPixelPrecisionBits, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(subTexelPrecisionBits, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(mipmapPrecisionBits, uint8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxDrawIndirectCount, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerLodBias, float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerAnisotropyLog2, uint8_t, 0.0f);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxViewports, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewportDims[0], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewportDims[1], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(viewportBoundsRange[0], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(viewportBoundsRange[1], float, 0.0f);
```

Comment:
similar to https://github.com/Devsh-Graphics-Programming/Nabla/pull/643/files#r1451616501